staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go

```diff
@@ -511,3 +512,43 @@ func TestIsFullyQualifiedName(t *testing.T) {
 		}
 	}
 }
+
+func TestIsEnvVarName(t *testing.T) {
+	successCases := []string{
+		"my.env-name",
+		"MY_ENV.NAME",
+		"MyEnvName1",
+		"simpleEnvName",
+		"now-Env-names",
+		"end-with-num-1",
+		"cantendwithadash-",
+	}
+	for i := range successCases {
+		t.Run(fmt.Sprintf("successCases string %s", successCases[i]), func(t *testing.T) {
+			if errs := IsEnvVarName(successCases[i]); len(errs) != 0 {
+				t.Errorf("case[%d] EnvVarName=%s expected success: %v", i, successCases[i], errs)
+			}
+		})
+	}
+
+	errorCases := []string{
+		".",
+		"..",
+		"..EnvName",
+		"nospecialchars%^=@",
+		"env/name/slash",
+		"env.name/abc",
+		"env_name/abc",
+		"example.com/",
+		"/envname",
+		"1-starts-with-num",
+		"",
```

Comment:
Add cases for colon and semicolon specifically please.  Those aren't allowed in bash shell per https://tiswww.case.edu/php/chet/bash/bash.html `name
A word consisting only of alphanumeric characters and underscores, and beginning with an alphabetic character or an underscore. Also referred to as an identifier.`

Probably worth linking to the man page too.